### PR TITLE
Make build script switch artifact IDs

### LINKED ISSUE #516: Project build fails after SBT upgrade to 1.9.0, but works for 1.8.3
## Expected Behavior (Mandatory)
Project compiles

## Actual Behavior (Mandatory)
SBT fails with the exception.
```
[info] Fetching artifacts of
[info] Fetched artifacts of
[error] java.net.URISyntaxException: Illegal character in path at index 123: https://repo1.maven.org/maven2/org/neo4j/neo4j-connector-apache-spark_2.12/5.0.0_for_spark_3/neo4j-connector-apache-spark_${scala.binary.version}-5.0.0_for_spark_3.jar
[error]         at java.net.URI$Parser.fail(URI.java:2847)
[error]         at java.net.URI$Parser.checkChars(URI.java:3020)
[error]         at java.net.URI$Parser.parseHierarchical(URI.java:3104)
[error]         at java.net.URI$Parser.parse(URI.java:3052)
[error]         at java.net.URI.<init>(URI.java:588)
[error]         at java.net.URL.toURI(URL.java:989)
[error]         at sjsonnew.JavaExtraFormats.$anonfun$urlStringIso$1(JavaExtraFormats.scala:40)
[error]         at sjsonnew.IsoString$$anon$1.to(IsoString.scala:26)
[error]         at sjsonnew.IsoFormats$$anon$2.write(IsoFormats.scala:31)
[error]         at sjsonnew.StandardFormats$OptionFormat.write(StandardFormats.scala:37)
[error]         at sjsonnew.StandardFormats$OptionFormat.addField(StandardFormats.scala:44)
[error]         at sjsonnew.StandardFormats$OptionFormat.addField(StandardFormats.scala:33)
[error]         at sjsonnew.Builder.addField(Builder.scala:43)
[error]         at sbt.librarymanagement.ArtifactFormats$$anon$1.write(ArtifactFormats.scala:36)
[error]         at sbt.librarymanagement.ArtifactFormats$$anon$1.write(ArtifactFormats.scala:9)
[error]         at sjsonnew.CollectionFormats$$anon$2.$anonfun$write$2(CollectionFormats.scala:85)
[error]         at sjsonnew.CollectionFormats$$anon$2.$anonfun$write$2$adapted(CollectionFormats.scala:85)
[error]         at scala.collection.Iterator.foreach(Iterator.scala:943)
[error]         at scala.collection.Iterator.foreach$(Iterator.scala:943)
[error]         at scala.collection.AbstractIterator.foreach(Iterator.scala:1431)
[error]         at scala.collection.IterableLike.foreach(IterableLike.scala:74)
[error]         at scala.collection.IterableLike.foreach$(IterableLike.scala:73)
[error]         at scala.collection.AbstractIterable.foreach(Iterable.scala:56)
[error]         at sjsonnew.CollectionFormats$$anon$2.write(CollectionFormats.scala:85)
[error]         at sjsonnew.CollectionFormats$$anon$2.write(CollectionFormats.scala:80)
[error]         at sjsonnew.JsonWriter.addField(JsonFormat.scala:40)
[error]         at sjsonnew.JsonWriter.addField$(JsonFormat.scala:37)
[error]         at sjsonnew.CollectionFormats$$anon$2.addField(CollectionFormats.scala:80)
[error]         at sjsonnew.Builder.addField(Builder.scala:43)
[error]         at sbt.librarymanagement.ModuleReportFormats$$anon$1.write(ModuleReportFormats.scala:42)
[error]         at sbt.librarymanagement.ModuleReportFormats$$anon$1.write(ModuleReportFormats.scala:9)
[error]         at sjsonnew.CollectionFormats$$anon$2.$anonfun$write$2(CollectionFormats.scala:85)
[error]         at sjsonnew.CollectionFormats$$anon$2.$anonfun$write$2$adapted(CollectionFormats.scala:85)
[error]         at scala.collection.Iterator.foreach(Iterator.scala:943)
[error]         at scala.collection.Iterator.foreach$(Iterator.scala:943)
[error]         at scala.collection.AbstractIterator.foreach(Iterator.scala:1431)
[error]         at scala.collection.IterableLike.foreach(IterableLike.scala:74)
[error]         at scala.collection.IterableLike.foreach$(IterableLike.scala:73)
[error]         at scala.collection.AbstractIterable.foreach(Iterable.scala:56)
[error]         at sjsonnew.CollectionFormats$$anon$2.write(CollectionFormats.scala:85)
[error]         at sjsonnew.CollectionFormats$$anon$2.write(CollectionFormats.scala:80)
[error]         at sjsonnew.JsonWriter.addField(JsonFormat.scala:40)
[error]         at sjsonnew.JsonWriter.addField$(JsonFormat.scala:37)
[error]         at sjsonnew.CollectionFormats$$anon$2.addField(CollectionFormats.scala:80)
[error]         at sjsonnew.Builder.addField(Builder.scala:43)
[error]         at sbt.librarymanagement.ConfigurationReportFormats$$anon$1.write(ConfigurationReportFormats.scala:26)
[error]         at sbt.librarymanagement.ConfigurationReportFormats$$anon$1.write(ConfigurationReportFormats.scala:9)
[error]         at sjsonnew.CollectionFormats$$anon$2.$anonfun$write$2(CollectionFormats.scala:85)
[error]         at sjsonnew.CollectionFormats$$anon$2.$anonfun$write$2$adapted(CollectionFormats.scala:85)
[error]         at scala.collection.Iterator.foreach(Iterator.scala:943)
[error]         at scala.collection.Iterator.foreach$(Iterator.scala:943)
[error]         at scala.collection.AbstractIterator.foreach(Iterator.scala:1431)
[error]         at scala.collection.IterableLike.foreach(IterableLike.scala:74)
[error]         at scala.collection.IterableLike.foreach$(IterableLike.scala:73)
[error]         at scala.collection.AbstractIterable.foreach(Iterable.scala:56)
[error]         at sjsonnew.CollectionFormats$$anon$2.write(CollectionFormats.scala:85)
[error]         at sjsonnew.CollectionFormats$$anon$2.write(CollectionFormats.scala:80)
[error]         at sjsonnew.JsonWriter.addField(JsonFormat.scala:40)
[error]         at sjsonnew.JsonWriter.addField$(JsonFormat.scala:37)
[error]         at sjsonnew.CollectionFormats$$anon$2.addField(CollectionFormats.scala:80)
[error]         at sjsonnew.Builder.addField(Builder.scala:43)
[error]         at sbt.librarymanagement.UpdateReportFormats$$anon$1.write(UpdateReportFormats.scala:27)
[error]         at sbt.librarymanagement.UpdateReportFormats$$anon$1.write(UpdateReportFormats.scala:9)
[error]         at sjsonnew.SupportConverter.toJsonUnsafe(SupportConverter.scala:23)
[error]         at sjsonnew.SupportConverter.toJsonUnsafe$(SupportConverter.scala:20)
[error]         at sjsonnew.support.scalajson.unsafe.Converter$.toJsonUnsafe(Converter.scala:7)
[error]         at sjsonnew.SupportConverter.$anonfun$toJson$1(SupportConverter.scala:14)
[error]         at scala.util.Try$.apply(Try.scala:213)
[error]         at sjsonnew.SupportConverter.toJson(SupportConverter.scala:14)
[error]         at sjsonnew.SupportConverter.toJson$(SupportConverter.scala:13)
[error]         at sjsonnew.support.scalajson.unsafe.Converter$.toJson(Converter.scala:7)
[error]         at sbt.util.FileOutput.write(Output.scala:38)
[error]         at sbt.util.FileBasedStore.write(CacheStore.scala:89)
[error]         at sbt.internal.InMemoryCacheStore$CacheStoreImpl.write(InMemoryCacheStore.scala:75)
[error]         at sbt.util.Tracked$.$anonfun$lastOutput$1(Tracked.scala:74)
[error]         at sbt.internal.LibraryManagement$.$anonfun$cachedUpdate$20(LibraryManagement.scala:146)
[error]         at scala.util.control.Exception$Catch.apply(Exception.scala:228)
[error]         at sbt.internal.LibraryManagement$.$anonfun$cachedUpdate$11(LibraryManagement.scala:146)
[error]         at sbt.internal.LibraryManagement$.$anonfun$cachedUpdate$11$adapted(LibraryManagement.scala:127)
[error]         at sbt.util.Tracked$.$anonfun$inputChangedW$1(Tracked.scala:219)
[error]         at sbt.internal.LibraryManagement$.cachedUpdate(LibraryManagement.scala:160)
[error]         at sbt.Classpaths$.$anonfun$updateTask0$1(Defaults.scala:3801)
[error]         at scala.Function1.$anonfun$compose$1(Function1.scala:49)
[error]         at sbt.internal.util.$tilde$greater.$anonfun$$u2219$1(TypeFunctions.scala:62)
[error]         at sbt.std.Transform$$anon$4.work(Transform.scala:68)
[error]         at sbt.Execute.$anonfun$submit$2(Execute.scala:282)
[error]         at sbt.internal.util.ErrorHandling$.wideConvert(ErrorHandling.scala:23)
[error]         at sbt.Execute.work(Execute.scala:291)
[error]         at sbt.Execute.$anonfun$submit$1(Execute.scala:282)
[error]         at sbt.ConcurrentRestrictions$$anon$4.$anonfun$submitValid$1(ConcurrentRestrictions.scala:265)
[error]         at sbt.CompletionService$$anon$2.call(CompletionService.scala:64)
[error]         at java.util.concurrent.FutureTask.run(FutureTask.java:266)
[error]         at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
[error]         at java.util.concurrent.FutureTask.run(FutureTask.java:266)
[error]         at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
[error]         at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
[error]         at java.lang.Thread.run(Thread.java:750)
[error] (update) java.net.URISyntaxException: Illegal character in path at index 123: https://repo1.maven.org/maven2/org/neo4j/neo4j-connector-apache-spark_2.12/5.0.0_for_spark_3/neo4j-connector-apache-spark_${scala.binary.version}-5.0.0_for_spark_3.jar
[error] Total time: 4 s, completed Jun 7, 2023 1:03:17 PM

```

## How to Reproduce the Problem
Update SBT to version `1.9.0` and try to compile the project using `neo4j-connector-apache-spark` as a dependency:
```
sbt clean compile
```

### Simple Dataset (where it's possible)
N/A

### Steps (Mandatory)
1. Create an SBT project
1. Add `libraryDependencies += "org.neo4j" %% "neo4j-connector-apache-spark" % "5.0.0_for_spark_3"`
1. Run SBT build

## Screenshots (where it's possible)
N/A

## Specifications (Mandatory)

Currently used versions

### Versions

- Spark: 3.3.2
- Scala: 2.12.15
- Neo4j:
- Neo4j Connector: 5.0.0_for_spark_3

## Additional information
N/A

### PR DESCRIPTION
**Pull request by fbiville**
Dynamic artifact IDs are not well supported by Maven

Fixes #516

### CODE REVIEW

**Review by ali-ince — APPROVED** [review]
LGTM.

**Review by conker84 — APPROVED** [review]

File: .github/workflows/python-ci.yml [hunk 1]
```diff
@@ -49,8 +49,11 @@ jobs:
         env:
           CI: true
         run: |
-          ./mvnw clean package -DskipTests -P scala-${{ matrix.spark-version.scala }} --no-transfer-progress
+          ./maven-release.sh package ${{ matrix.spark-version.scala }}
+          head -n 15 spark-3/pom.xml
       - name: Run tests for Spark ${{ matrix.spark-version.ext }} and Neo4j ${{ matrix.neo4j-version }}
         run: |
+          pwd
+          head -n 15 spark-3/pom.xml
```
Comment (conker84) [inline .github/workflows/python-ci.yml]: @fbiville Do we need this?

Comment (fbiville) [reply .github/workflows/python-ci.yml]: nope, will remove it now that it finally works

**Comment by fbiville** [comment]
# How to test this

## SBT project

Install Scala (the rest of the instructions assumes a Scala 2.13.8 version)
Install SBT (`sbt --version` should show 1.9.0).
Create an empty custom Maven local repository, here `/tmp/custom-local-repo`.

Create a SBT project structured as follows.

- `build.sbt`

```scala
ThisBuild / scalaVersion := "2.13.8"
ThisBuild / organization := "com.example"

lazy val hello = (project in file("."))
  .settings(
    name := "Hello",
    libraryDependencies += "org.neo4j" %% "neo4j-connector-apache-spark" % "5.0.999_for_spark_3"

  )

resolvers += MavenCache("local-maven", file("/tmp/custom-local-repo"))
```

 - `src/main/scala/example/Hello.scala`
 
```scala
package example

object Hello {
  def main(args: Array[String]): Unit = {
    println("Hello")
  }
}
```

## Before

First, set a custom, non-released version (`5.0.999`).

```shell
cd /path/to/neo4j-spark-connector
git checkout 5.0
mvn versions:set -DnewVersion=5.0.999
./maven-release.sh deploy 2.13 local::file:///tmp/custom-local-repo
```

Now, try to run the SBT project:

```shell
cd /path/to/sbt-project
sbt run
```

You should see something like:
```java
[error] java.net.URISyntaxException: Illegal character in path at index 111: file:/tmp/barfoo/org/neo4j/neo4j-connector-apache-spark_2.13/5.0.999_for_spark_3/neo4j-connector-apache-spark_${scala.binary.version}-5.0.999_for_spark_3.ja
```

Before moving on, remove all contents of `/tmp/custom-local-repo` and run `git checkout .` on the spark connector repo.

## After

Set a custom, different non-released version (`5.0.99999`):

```shell
cd /path/to/neo4j-spark-connector
git checkout issue_516
mvn versions:set -DnewVersion=5.0.99999
./maven-release.sh deploy 2.13 local::file:///tmp/custom-local-repo
```

Update `build.sbt` to replace `5.0.999_for_spark_3` with `5.0.99999_for_spark_3`.

Now, run the SBT project:

```shell
cd /path/to/sbt-project
sbt run
```

You should see something like:
```shell
[info] running example.Hello
Hello
```

**Review by ali-ince — APPROVED** [review]

**Comment by fbiville** [comment]
Dry-run release in CI worked fine.

**Pull request merged**